kedro/framework/cli/starters.py

```diff
@@ -38,6 +43,81 @@
 }
 _STARTERS_REPO = "git+https://github.com/kedro-org/kedro-starters.git"
 
+
+@define(order=True)
+class KedroStarterSpec:
+    """Specification of custom kedro starter template.
+
+    Args:
+        name: name of the starter (name of the directory)
+        template_path: the path to git repository
+        directory: by default it will look for `name`, optionally override with custom directory.
+        origin: preserved field used by kedro internally to determine where does the starter
+        come from, user do not need to provide this field.
+
+    Example:
+    ::
+        >>> spec = KedroStarterSpec(
+                        name="astro-airflow-iris",
+                        template_path="git+https://github.com/kedro-org/kedro-starters.git",
+                        directory=None
+                                   )
+    """
+
+    name: str
+    template_path: str
+    directory: Optional[str] = None
+    origin: Optional[str] = field(init=False, default=None)
```

Comment:
```suggestion
    origin: Optional[str] = field(init=False, default="kedro")
```
Just to remove the need for L118?